nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -34,68 +35,35 @@
 from nemo.collections.nlp.modules.common.tokenizer_utils import get_nmt_tokenizer
-from nemo.collections.nlp.modules.common.transformer.text_generation import (
-    LengthParam,
-    OutputType,
-    SamplingParam,
-    TextGeneration,
-)
+from nemo.collections.nlp.modules.common.transformer.text_generation import LengthParam, OutputType, SamplingParam
 from nemo.collections.nlp.parts.nlp_overrides import GradScaler
 from nemo.collections.nlp.parts.utils_funcs import get_last_rank
 from nemo.core.classes.common import PretrainedModelInfo
 from nemo.core.optim import MainParamsOptimizerWrapper, prepare_lr_scheduler
 from nemo.utils import AppState, logging
 
 try:
-    from apex.transformer import parallel_state
-    from apex.contrib.layer_norm.layer_norm import FastLayerNorm
-    from apex.normalization.fused_layer_norm import FusedLayerNorm  # NOQA
-    from apex.transformer.pipeline_parallel.schedules.common import (
-        build_model,
-        listify_model,
-    )
+    from apex.transformer import parallel_state, tensor_parallel
+    from apex.transformer.pipeline_parallel.schedules.common import build_model
     from apex.transformer.pipeline_parallel.schedules.fwd_bwd_pipelining_without_interleaving import (
         forward_backward_pipelining_without_interleaving,
     )
     from apex.transformer.pipeline_parallel.schedules.fwd_bwd_no_pipelining import forward_backward_no_pipelining
-    from apex.transformer.pipeline_parallel.utils import get_num_microbatches
+    from apex.transformer.pipeline_parallel.utils import get_num_microbatches, _reconfigure_microbatch_calculator
 
     HAVE_APEX = True
 except (ImportError, ModuleNotFoundError):
     HAVE_APEX = False
 
 
-def _get_params_for_weight_decay_optimization(
-    model: Union[torch.nn.Module, List[torch.nn.Module]],
-) -> Dict[str, torch.nn.Parameter]:
-    """Divide params into with-weight-decay and without-weight-decay groups.
-
-    Layernorms and biases will have no weight decay but the rest will.
-    """
-    modules = listify_model(model)
-    weight_decay_params = {'params': []}
-    no_weight_decay_params = {'params': [], 'weight_decay': 0.0}
-    for module in modules:
-        for module_ in module.modules():
-            if isinstance(module_, (FusedLayerNorm, FastLayerNorm)):
-                no_weight_decay_params['params'].extend(
-                    [p for p in list(module_._parameters.values()) if p is not None]
-                )
-            else:
-                weight_decay_params['params'].extend(
-                    [p for n, p in list(module_._parameters.items()) if p is not None and n != 'bias']
-                )
-                no_weight_decay_params['params'].extend(
-                    [p for n, p in list(module_._parameters.items()) if p is not None and n == 'bias']
-                )
-
-    return weight_decay_params, no_weight_decay_params
```

Comment:
Why is this deleted?

Reply:
It moved to a different file and is imported from there.

Reply:
This is because I needed it for T5 as well.

Reply:
Thanks.